crates/starfish/core/src/synchronizer.rs

```diff
@@ -2548,10 +2673,88 @@ mod tests {
         // 10) Fourth fetch from peer 5 (fallback after peer 2 timeout)
         let (_guard5, bytes5, peer5) = &results[3];
         assert_eq!(*peer5, AuthorityIndex::new_for_test(5));
-        let expected5 = missing_verified_block_headers[0..known_number_block_headers]
+        let expected5 = all_verified_block_headers[0..known_number_block_headers]
             .iter()
             .map(|vb| vb.serialized().clone())
             .collect::<Vec<_>>();
         assert_eq!(bytes5, &expected5);
     }
+
+    #[tokio::test]
+    async fn test_process_fetched_headers_with_future_timestamp() {
```

Comment:
Please add more comments describing what is tested and the expected outcome, as it's a bit unclear. If we wait a bit longer then would the other block with the future timestamp be added as well? Or is it ignored completely? The comments in the test suggest that both blocks are added (` // THEN ensure those ended up in Core` ) but we assert that only one has been added.